raiden/utils/http.py

```diff
@@ -124,6 +124,42 @@ def start(self):
             raise
         return self
 
+    def kill(self):
+        STDOUT = subprocess.STDOUT  # pylint: disable=no-member
+
+        ps_fax = subprocess.check_output(["ps", "fax"], stderr=STDOUT)
+
+        log.debug("Executor process: killing process", command=self.command)
+        log.debug("EXecutor process: current processes", ps_fax=ps_fax)
+
+        super().kill()
+
+        ps_fax = subprocess.check_output(["ps", "fax"], stderr=STDOUT)
+
+        log.debug("Executor process: process killed", command=self.command)
+        log.debug("EXecutor process: current processes", ps_fax=ps_fax)
+
+    def wait_for(self, wait_for):
+        while self.check_timeout():
+            log.debug(
+                "Executor process: waiting",
+                command=self.command,
+                until=self._endtime,
+                now=time.time(),
+            )
+            if wait_for():
+                log.debug(
+                    "Executor process: waiting ended",
+                    command=self.command,
+                    until=self._endtime,
+                    now=time.time(),
+                )
+                return self
```

Comment:
Why not just `return`? The overridden method does not return `self`: https://mirakuru.readthedocs.io/en/v0.6.0/_modules/mirakuru/base.html#SimpleExecutor.wait_for

Reply:
That is not the behavior of the version we are currently using. For `mirakuru==1.1.0` it [returns self](https://github.com/ClearcodeHQ/mirakuru/blob/0c1093eb1c2ce76722364646229d5ffa64d77949/src/mirakuru/base.py#L349). I literally copied and pasted the code.